src/main/java/in/org/projecteka/hiu/user/UserController.java

```diff
@@ -32,10 +33,29 @@
                 .map(user -> user.toBuilder()
                         .password(passwordEncoder.encode(user.getPassword()))
                         .role(user.getRole() == null ? Role.DOCTOR : user.getRole())
+                        .verified(false)
                         .build())
                 .flatMap(userRepository::save);
     }
 
+    @PutMapping("/users/password")
+    public Mono<Void> changePassword(@RequestBody ChangePasswordRequest changePasswordRequest) {
+        var passwordValidation = PasswordValidator.validate(changePasswordRequest);
+        return passwordValidation.isValid() ?
```

Comment:
it could be simpler by validating the failure scenario first.

if(!passwordValidation.isValid)
{
return Mono.error(new ClientError(BAD_REQUEST, new ErrorRepresentation(new Error(INVALID_REQUEST, passwordValidation.getError()))));
}

return otherwise;